scripts/build_char2array.py

```diff
@@ -0,0 +1,105 @@
+import argparse
+import pickle
+from os import listdir
+import numpy as np
+from src.utils.common import text2image
+from fontTools.ttLib import TTFont
+from loguru import logger
+from tqdm import tqdm
+import ftfy
+
+
+def text2numpy(text: str, font: str, font_size: int = 16):
+    image = text2image(text, font=font, font_size=font_size)
+    image_bytes = np.array(image)
+    image_bytes = 255 - image_bytes
+    return image_bytes
+
+
+# Check whether font supports symbol
+def has_glyph(font: TTFont, glyph: str) -> bool:
+    assert len(glyph) == 1, glyph
+    for table in font["cmap"].tables:
+        if ord(glyph) in table.cmap.keys():
+            return True
+    return False
+
+
+# Find correct font that supports symbol or returns None
+def pick_font(symbol: str, fonts: list) -> str | None:
+    for font in fonts:
+        if has_glyph(TTFont(font), symbol):
+            return font
+    return None
+
+
+def build_char2array(
+    save_to: str,
+    fonts_directory: str,
+    chars_path: str,
+    font_size: int = 16,
+    fill_value_unknown: int = 255,
+    width_unknown: int = None,
+    ignore_invalid_encoding: bool = False,
+):
+    assert fill_value_unknown in range(256), "Value of filling must be int number in [0,255]"
+    fonts = [f"{fonts_directory}/{font}" for font in listdir(fonts_directory) if font.endswith(("ttf", "otf"))]
+
+    with open(chars_path, "rb") as f:
+        all_chars = pickle.load(f)
+
+    char2array = dict()
+    not_supported = []
+    problem_chars = []
+    for char in tqdm(all_chars):
+        current_font = pick_font(char, fonts)
+
+        if current_font:
+            try:
+                char2array[char] = text2numpy(char, font=current_font, font_size=font_size)
+            except:
```

Comment:
It is not good to control execution flow with `try..catch` statements. Moreover, it is bad to catch all exceptions, because you may face something unrelated to the script (e.g. something from OS), and you wouldn't know about this.

What kind or errors may occur here?

Reply:
I double checked that we really don't catch exceptions